spec/lib/search_spec.rb

```diff
@@ -2580,4 +2580,34 @@ def indexed_post(*args)
       expect(result.categories.length).to eq(0)
     end
   end
+
+  context "when max_duplicate_search_index_terms limits duplication" do
```

Comment:
I've added a test to `search_indexer_spec` below and I personally feel like this testing too many factors (ranking, weights) for the change we're making here.

Reply:
test looks good!